### ISSUE TITLE
Provide adequate overrides to System settings on a per data provider level.

### ISSUE THREAD

**Issue by StephenHodgson** [XRTK/XRTK-Core#294]
# XRTK - Mixed Reality Toolkit Feature Request

## Is your feature request related to a problem? Please describe

When platform level data provider configurations are changed from the default setting in the parent service level's configuration profile, then we should be properly supplying an "override" to that data provider with the setting.

The best example of this atm is the Spatial Awareness System and the WMR/Lumin data providers. There's global settings in the spatial awareness system that can be adjusted for all, then a per platform setting can also be supplied that will effectively override the global setting.

## How would you classify your suggestion

- Usability / Configuration
- Architecture / Services

## Acceptance Criteria

- [ ] Developers can set global settings for a service using it's configuration profile
- [ ] Developers can set per platform settings using a data provider's configuration profile on the parent service.

**Comment by SimonDarksideJ** [comment]
My main issue with setting "Per Service" platform level overrides, is that you are taking an already heavy configuration layout and making it far more complicated.
Also if I want to look at the configuration for a single platform, I now have to hunt through each service to find the differences.  The approach in #290 is far simplier, bu grouping each platforms config under a single header.

They can still chare configuration, but allows the developer to customise a "Set" of config for a platform.

**Comment by StephenHodgson** [comment]
You are setting a custom config per platform, on the data provider level.

The top down approach means that each master configuration can me wildly different than the per platform configuration.

**Comment by StephenHodgson** [comment]
I suppose it's simpler in the fact that you can easily see the changes, but it's more complex because of the vast differences you can have form one master configuration to another.

Where as the bottom up approach means that the platform is configured exactly as it should be and all the other settings are shared between all platforms.

**Comment by SimonDarksideJ** [comment]
Settings can still be shared between platforms, I would simply create folders for all my config.  Putting the global (shared) config in one folder.  Then creating platform variances only where I needed them

**Comment by StephenHodgson** [comment]
Imho, it' simply a UX problem. But in reality we really need it to be this way from a arch standpoint.

We can easily make a window that shows just the specific platform settings.

**Comment by StephenHodgson** [comment]
> Putting the global (shared) config in one folder. Then creating platform variances only where I needed them

That's already the way it works

**Comment by SimonDarksideJ** [comment]
Yes, except currently, there is no way to separate config per platform. Other than in the Data providers.
It's not available for the core services, etc.

**Comment by StephenHodgson** [comment]
Core services are the global shared config, and the data provider profiles are the per platform settings.

We just need a way to make this easier to access via UX updates.

**Comment by StephenHodgson** [comment]
And make it more obvious that the data provider settings take precedent (even tho it may be the same options on the system level.

**Comment by StephenHodgson** [comment]
TL;DR I don't wanna change the way the system fundamentally works, when it's just a UX problem.

**Comment by SimonDarksideJ** [comment]
Can you update the discussion in #290 and close this FR (since they are the same thing).  Post your alternate proposal there.

**Comment by StephenHodgson** [comment]
I just wanted to keep it clean by opening a new issue as an alt. Linked back here, just in case.

Is that okay?
